_posts/2017-05-26-jens-gheerardyn-the-blockchain-how-does-it-work.md

```diff
@@ -0,0 +1,104 @@
+</div>
+<div class="1u">&nbsp;</div>
+<div class="7u">
+In order to prevent tampering with the data we need some sort of validation. A hash or more specific a cryptographic hash function, such a function maps data from an arbitrary size to a fixed size. Such a function is also called a non-ivertible or a one-way function. The input will always generate the same output but you cannot figure out the input from the output, hence the term one-way. 
+
+Sha256 is a hashing algorithm with an output of 256 bits or 64 hexadecimal characters. A Hexadecimal character uses 4 bits per character: 256/4=64
+
+{% include image.html img="/img/blockchain/sha256.png" alt="Hash algorithm - variable size to fixed size" title="Hash algorithm - variable size to fixed size" %}
+
+The hash for ncore will always be the same. Even an empty value has a hash value.
+
+{% include image.html img="/img/blockchain/sha256-empty.png" %}
+
+Let's take a very long Lorem Ipsum and see what happens
+
+{% include image.html img="/img/blockchain/sha256-lorem-ipsum.png" %}
+
+The hash has always the same length no matter the input.
+</div>
+</div>
+
+
+# Block
+Creating just a hash from some input is not very complicated and has no real added value. Where the added value comes from is that there is a constraint. The constraint makes it harder to create a hash. When we look at the constraint from Bitcoin today we will see that the all the hashes have 16 leading 0's.
+
+Given some input we get an output, now we want to make the output more complicated therefore we need to add a nonce. For our example we add the constraint that the nonce should has 4 leading 0's.
+
+{% include image.html img="/img/blockchain/block-empty.png" %}
+
+On an empty input everything is okay, now let's find out what happens when we change the input to NCore.
+
+{% include image.html img="/img/blockchain/block-ncore.png" %}
+
+The moment we changed the data to NCore the hash violated the constraint and became invalid. In order to fix this, we need to find a matching nonce. You can probably guess why there is a "Mine" button at the bottom.
+
+{% include image.html img="/img/blockchain/block-ncore-mined.png" %}
+
+# Blockchain
+A blockchain is a chain of blocks. The current block takes the hash of the previous block. The chain breaks when you to modify a value, this is exactly why they describe the blockchain as immutable. It is append only.
+
+{% include image.html img="/img/blockchain/blockchain.png" %}
+
+# Distributed
+If you had full control over the Blockchain it would only be a matter of time until you had altered an entry and cheated the system. Therefore the blockchain is also distributed this means that there must be an agreement for every new entry. When you would try to modify something from the past the chain would be broken. 
```

Comment:
Therefore the blockchain is also distributed; this means that there must be an agreement for every new entry.